### PR TITLE
feat: [CCM-10902]: Cloud-Functions' changes for Cost-Category support in Looker

### PR DESCRIPTION
**Pull request by abhijeet-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43763)

### CODE REVIEW

**Review by trunapushpa — DISMISSED** [review]

**Comment by abhijeet-harness** [comment]
trigger ti0

File: scripts/terraform/cloudfunctions/ce/src/python/aws_billing_bq_main.py [hunk 1]
```diff
@@ -506,10 +521,11 @@ def create_dataset_and_tables(jsonData):
...
             # Enable these only when needed.
-            # if table_ref == aws_cur_table_ref:
-            #     alter_awscur_table(jsonData)
-            # elif table_ref == unified_table_ref:
-            #     alter_unified_table(jsonData)
+            if table_ref == aws_cur_table_ref:
+                # alter_awscur_table(jsonData)
+                pass
+            elif table_ref == unified_table_ref:
+                alter_unified_table(jsonData)
```
Comment (nikunj-badjatya) [inline scripts/terraform/cloudfunctions/ce/src/python/aws_billing_bq_main.py:528]: want to call unified table alter always ?

File: scripts/terraform/cloudfunctions/ce/src/python/aws_billing_bq_main.py [hunk 2]
```diff
@@ -958,8 +976,10 @@ def ingest_data_to_costagg(jsonData):
 def alter_unified_table(jsonData):
     print_("Altering unifiedTable Table")
     ds = "%s.%s" % (PROJECTID, jsonData["datasetName"])
+    # query = "ALTER TABLE `%s.unifiedTable` \
+    #     ADD COLUMN IF NOT EXISTS awsBillingEntity STRING;" % ds
     query = "ALTER TABLE `%s.unifiedTable` \
-        ADD COLUMN IF NOT EXISTS awsBillingEntity STRING;" % ds
+        ADD COLUMN IF NOT EXISTS costCategory ARRAY<STRUCT<costCategoryName STRING, costBucketName STRING>>;" % ds
```
Comment (nikunj-badjatya) [inline scripts/terraform/cloudfunctions/ce/src/python/aws_billing_bq_main.py:982]: This should be there in bq_schema.py as well

**Review by nikunj-badjatya — APPROVED** [review]

File: scripts/terraform/cloudfunctions/ce/src/python/aws_billing_bq_main.py (hunk 1 above)
Comment (abhijeet-harness) [reply scripts/terraform/cloudfunctions/ce/src/python/aws_billing_bq_main.py]: Disabled alter statements. Moved it to a one time alter-query execution for all accounts.

File: scripts/terraform/cloudfunctions/ce/src/python/aws_billing_bq_main.py (hunk 2 above)
Comment (abhijeet-harness) [reply scripts/terraform/cloudfunctions/ce/src/python/aws_billing_bq_main.py:980]: Added.

**Review by rdsoze — APPROVED** [review]

**Pull request merged**